### PR TITLE
Add documentation

### LINKED ISSUE #1: Add documentation

### PR DESCRIPTION
**Pull request by ilyavy**
Closes #1, #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,601 @@
...
+
+#### Line breaks with operators and expressions
+If the expression's length exceeds **120 symbols**, delimiters (commas, operators, etc.) should be left at the end of the line, 
+and not be moved to the next line. Exception - in the method call, like *Builder Pattern*, *Java Streams*.
+In lambda expressions `-> {` should be left on the first line.
+
+##### if example
+Correct:
+```java
+if (veryImportantFlagWithLongName == true && theLineIsLongerThanShouldBe == true &&
```
Comment (aaukhatov) [inline README.md:296]: I've changed my mind. Vice versa, if you don't mind. I mean with the leading operator.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,601 @@
...
+ * Creates a concrete book file reader by the specified file path. A concrete implementation is chosen based
+ * on the file's extension.
+ *
+ * @param bookFile file referencing a book
+ * @return BookFileReader instance
+ * @throws IllegalArgumentException      if the specified file does not exist
+ * @throws UnsupportedOperationException if an extension of the file is not supported
+ * @see TxtFileReader
+ */
+public static BookFileReader createInstance(final File bookFile) {
```
Comment (aaukhatov) [inline README.md:528]: Please, don't use the `final` keyword in a method signature.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,601 @@
...
+##### Unit Tests
+Unit test class names should be ended with postfix `Test`.
+
+Example of unit test on class `Cat.java`:
+```
+CatTest.java
+```
+
+#### Integration Tests
+Integration test class name should end with postfix `IT`.
```
Comment (aaukhatov) [inline README.md:565]: I suppose we can allow `IntegrationTest` postfix too.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,601 @@
...
+public class Cat extends
+    Animal {
+
+}
+```
+
+Incorrect:
+```java
+public final class Cat 
+    extends Animal {
```
Comment (aaukhatov) [inline README.md:406]: For me this case better to read

**Review by aaukhatov — CHANGES_REQUESTED** [review]
@ilyavy I'm assuming, we should show a way how to suppress particular rules too.
What do you think?

File: README.md [hunk 5]
```diff
@@ -1,3 +1,618 @@
...
+5. _Empty line_
+6. One high-level class
+
+### License
+The license should be put in each source code file in a header.
+
+#### An example of full license text
+```java
+ /*
+ * Copyright (C) 2021 Arthur Aukhatov
```
Comment (aaukhatov) [inline README.md:174]: I really appreciate it)
But we should change it to **Pepper Kit**

File: README.md [hunk 6]
```diff
@@ -1,3 +1,618 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ ```
+
+#### An example of shortened license text
+ ```java
+/*
+ * Copyright (C) 2021 Arthur Aukhatov
```
Comment (aaukhatov) [inline README.md:193]: **Pepper Kit**

File: README.md (hunk 1 above)
Comment (ilyavy) [reply README.md:313]: Ok, thought about it too :)

File: README.md (hunk 4 above)
Comment (ilyavy) [reply README.md:423]: Ok, I'll remove this rule then :)

File: README.md (hunk 5 above)
Comment (ilyavy) [reply README.md:174]: Ok :)

File: README.md (hunk 1 above)
Comment (ilyavy) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (ilyavy) [reply README.md:543]: done

File: README.md (hunk 4 above)
Comment (ilyavy) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (ilyavy) [reply README.md:173]: done

File: README.md (hunk 6 above)
Comment (ilyavy) [reply README.md:192]: done

**Review by aaukhatov — APPROVED** [review]

**Pull request merged**